packages/toc-extension/src/index.ts

```diff
@@ -42,212 +52,196 @@ namespace CommandIDs {
-  if (markdownViewerTracker) {
-    const renderedMarkdownGenerator = createRenderedMarkdownGenerator(
-      markdownViewerTracker,
-      toc,
-      rendermime.sanitizer,
-      translator,
-      settings
-    );
-    registry.add(renderedMarkdownGenerator);
-  }
+      // Create a LaTeX generator:
+      const latexGenerator = createLatexGenerator(editorTracker);
+      registry.add(latexGenerator);
 
-  // Update the ToC when the active widget changes:
-  if (labShell) {
-    labShell.currentChanged.connect(onConnect);
-  }
+      // Create a Python generator:
+      const pythonGenerator = createPythonGenerator(editorTracker);
+      registry.add(pythonGenerator);
+    }
 
-  return registry;
-
-  /**
-   * Callback invoked when the active widget changes.
-   *
-   * @private
-   */
-  function onConnect() {
-    let widget = app.shell.currentWidget;
-    if (!widget) {
-      return;
+    // Create a rendered Markdown generator:
+    if (markdownViewerTracker) {
+      const renderedMarkdownGenerator = createRenderedMarkdownGenerator(
+        markdownViewerTracker,
+        toc,
+        rendermime.sanitizer,
+        translator,
+        settings
+      );
+      registry.add(renderedMarkdownGenerator);
     }
-    let generator = registry.find(widget);
-    if (!generator) {
-      // If the previously used widget is still available, stick with it.
-      // Otherwise, set the current ToC widget to null.
-      if (toc.current && toc.current.widget.isDisposed) {
-        toc.current = null;
+
+    const updateCurrent = () => {
+      const widget = app.shell.currentWidget;
+      if (!widget) {
+        return;
       }
-      return;
+      const generator = registry.find(widget);
+      if (!generator) {
+        // If the previously used widget is still available, stick with it.
+        // Otherwise, set the current ToC widget to null.
+        if (toc.current && toc.current.widget.isDisposed) {
+          toc.current = null;
+        }
+        return;
+      }
+      toc.current = { widget, generator };
+    };
+
+    // Update the ToC when the active widget changes
+    if (labShell) {
+      labShell.currentChanged.connect(updateCurrent);
     }
-    toc.current = { widget, generator };
-  }
-}
 
-/**
- * Initialization data for the ToC extension.
- *
- * @private
- */
-const extension: JupyterFrontEndPlugin<ITableOfContentsRegistry> = {
-  id: '@jupyterlab/toc:plugin',
-  autoStart: true,
-  provides: ITableOfContentsRegistry,
-  requires: [IDocumentManager, IRenderMimeRegistry, ITranslator],
-  optional: [
-    IEditorTracker,
-    ILayoutRestorer,
-    ILabShell,
-    IMarkdownViewerTracker,
-    INotebookTracker,
-    ISettingRegistry
-  ],
-  activate: activateTOC
+    app.started.then(updateCurrent);
+
+    return registry;
+  }
 };
 
 /**
- * Exports.
+ * Export the plugins as default.
  */
-export default extension;
+const plugins: JupyterFrontEndPlugin<any>[] = [registry];
+
+export default plugins;
```

Comment:
Minor optional suggestion, we don't need to export an array.

```suggestion
export default registry;
```

Reply:
Sure. I was looking into splitting the main plugin into multiple ones at first, but happy to switch to a more concise export.